src/components/Board.vue

```diff
@@ -1,89 +1,129 @@
-    const grMinHeight = 90
-    const grMaxHeight = 60
-    const goodMinHeight = 110
-    const goodMaxHeight = 30
-    const greatArea = itemPosition >= this.gameFlowHeight - grMinHeight && itemPosition <= this.gameFlowHeight - grMaxHeight
-    const goodArea = itemPosition >= this.gameFlowHeight - goodMinHeight && itemPosition <= this.gameFlowHeight - goodMaxHeight
-
-    if (greatArea) {
-      this.scoreCounter += 2
-    } else if (goodArea) {
-      this.scoreCounter += 1
+  isPlay: boolean = false
+  safeLoop: number = 0
+
+  get storeItems (): IArrowData[] {
+    return this.$store.state.arrowStore.arrowsData
+  }
+
+  get positions (): IFlowProps {
+    const goodArea: any = this.$refs.good
+    const exArea: any = this.$refs.excellent
+    return {
+      goodArTop: goodArea.getBoundingClientRect().top,
+      goodArBottom: goodArea.getBoundingClientRect().bottom,
+      exAreaTop: exArea.getBoundingClientRect().top,
+      exAreaBottom: exArea.getBoundingClientRect().bottom
     }
-  this.scoreElement!.textContent = this.scoreCounter.toString()
   }
 
-  delay = (ms: number) => {
-    return new Promise(resolve => setTimeout(resolve, ms))
+  startGame (): void {
+    console.log('the game is on')
+    this.isPlay = true
+    this.runRender()
+    window.addEventListener('keydown', this.logKey)
   }
 
-  renderBlocks = async (elements: any) => { // function will be deleted
-    for (let i = 0; i < elements.length; i += 1) {
-      if (i === 0) {
-        this.gameFlow?.appendChild(elements[i]) // components will be added using VUE methods
+  runRender (): void {
+    const interval = setInterval(() => {
+      this.$store.dispatch(`${StoreModuleEnum.arrowStore}/${EActionArrow.ADD_DATA}`)
+      this.safeLoop += 1
+      clearInterval(interval)
+      if (this.safeLoop < 1000 && this.isPlay) {
+        this.runRender()
       } else {
-        await this.delay(this.block.getRandom().timeout)
-        this.gameFlow?.appendChild(elements[i])
+        console.log('stop')
+        setTimeout(() =>
+          window.removeEventListener('keydown', this.logKey),
+        5000)
       }
+    }, this.getRandom(200, 2000))
+  }
 
-      elements[i].addEventListener('animationend', () => {
-        document.getElementById(elements[i].id)?.remove() // components will be deleted using VUE methods
-      })
-    }
+  stopGame (): void {
+    this.isPlay = false
   }
 
-  startGame () {
-    this.counts += 1
-    console.log('the game is on')
-    const elements = []
-    for (let i = 1; i <= this.block.getRandom().elements; i += 1) {
-      elements.push(new Block().createItem())
+  logKey (e: KeyboardEvent): void {
+    const key: string = e.key
+    const keyMap: { [index: string]: EDirection } = {
+      ArrowLeft: EDirection.ArrowLeft,
+      ArrowUp: EDirection.ArrowUp,
+      ArrowDown: EDirection.ArrowDown,
+      ArrowRight: EDirection.ArrowRight
+    }
+    const singleEl = this.$store.state.arrowStore.arrowsData.find((e: IArrowData) =>
```

Comment:
`find` returns the first arrow with a direction that is the same as a user clicked